app/services/reporter.rb

```diff
@@ -6,61 +6,80 @@
     druid_array
   end
 
-  # @param [Array] druids - list of druids to output details for
   # @param [Boolean] errors_only (default: false) - optionally only output lines with audit errors
   # @return [Array] an array of hashes with details for each druid provided
-  def moab_detail_for(druids, errors_only: false)
-    detail_array = []
-    druids.each do |druid|
-      preserved_object = PreservedObject.find_by(druid: druid)
-      preserved_object.complete_moabs.each do |cm|
-        next if errors_only && cm.status == 'ok'
-        detail_array << { druid: druid,
-                          status: cm.status,
-                          status_details: cm.status_details,
-                          last_moab_validation: cm.last_moab_validation,
-                          last_checksum_validation: cm.last_checksum_validation,
-                          storage_root: cm.moab_storage_root.name,
-                          from_storage_root: cm.from_moab_storage_root&.name }
+  def moab_detail_csv_list(errors_only: false)
+    query =
+      if errors_only
+        moab_storage_root_list_pres_obj_comp_moabs_relation.where.not(complete_moabs: { status: 'ok' })
+      else
+        moab_storage_root_list_pres_obj_comp_moabs_relation
       end
+
+    header_row = [['druid', 'from_storage_root', 'storage_root', 'last_checksum_validation', 'last_moab_validation', 'status', 'status_details']]
+
+    # cols doesn't include storage_root name (available via storage_root ivar).  also we're not instantiating AR objects, so we have to translate
+    # the underlying status enum
+    cols = ['druid', 'from_moab_storage_root_id', 'last_checksum_validation', 'last_moab_validation', 'status AS status_code', 'status_details']
+    data_rows = query.select(cols).each_row.map do |po_cm_hash|
+      [
+        po_cm_hash['druid'], moab_storage_root_name(po_cm_hash['from_moab_storage_root_id']), storage_root.name,
+        po_cm_hash['last_checksum_validation'], po_cm_hash['last_moab_validation'], status_text_from_code(po_cm_hash['status_code']),
+        po_cm_hash['status_details']
+      ]
     end
-    detail_array
+    header_row + data_rows
+  end
+
+  def moab_storage_root_name(msr_id)
+    return nil unless msr_id.present?
+    @msr_names[msr_id] ||= MoabStorageRoot.find(msr_id).name
+  end
+
+  def status_text_from_code(status_code)
+    # statuses is a hash of the form { 'status_text' => status_code }.  this is just a reverse hash lookup.
+    CompleteMoab.statuses.to_a.find { |s| s[1] == status_code }[0]
   end
 
   # @param [Array] lines - values to output on each line of the csv
   # @param [String] filename - optional filename to override the default
   # @return [String] the name of the CSV file to which the list was written
-  def write_to_csv(lines, filename: nil)
-    filename ||= default_filename(filename_prefix: "MoabStorageRoot_#{storage_root.name}_druids", filename_suffix: 'csv')
+  def write_to_csv(lines, report_type: nil, filename: nil)
```

Comment:
same comment about params hash;  I think positional params are fine for the context of this method.